tests/components/solax/test_config_flow.py

```diff
@@ -0,0 +1,138 @@
+"""Tests for the solax config flow."""
+from unittest.mock import patch
+
+from solax import RealTimeAPI, inverter
+from solax.inverter import InverterResponse
+
+from homeassistant import config_entries, setup
+from homeassistant.components.solax.const import DOMAIN
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PASSWORD, CONF_PORT
+
+
+def __mock_real_time_api_success():
+    return RealTimeAPI(inverter.X1MiniV34)
+
+
+def __mock_get_data():
+    return InverterResponse(
+        data=None, serial_number="ABCDEFGHIJ", version="2.034.06", type=4
+    )
+
+
+async def test_form_success(hass):
+    """Test successful form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    flow = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert flow["type"] == "form"
+    assert flow["errors"] == {}
+
+    with patch(
+        "homeassistant.components.solax.config_flow.real_time_api",
+        return_value=__mock_real_time_api_success(),
+    ), patch("solax.RealTimeAPI.get_data", return_value=__mock_get_data()), patch(
```

Comment:
You probably want to patch the client in the config flow and not the global client.

Reply:
Not sure I understood. Do you mean the get_data call from the API? Or the full validate_api function?

I changed to this and it works, even though I think it tests less functionality this way:
```
with patch(
        "homeassistant.components.solax.config_flow.validate_api",
        return_value="ABCDEFGHIJ",
    )
```

Reply:
Aah nevermind, yeah I see this issue, this is the correct way to do it here indeed